core/capabilities/compute/compute.go

```diff
@@ -270,25 +342,40 @@ func (c *Compute) createFetcher() func(ctx context.Context, req *wasmpb.FetchReq
 	}
 }
 
+const (
+	defaultNumWorkers = 3
+)
+
+type Config struct {
+	webapi.ServiceConfig
+	NumWorkers int
+}
+
 func NewAction(
-	config webapi.ServiceConfig,
+	config Config,
 	log logger.Logger,
 	registry coretypes.CapabilitiesRegistry,
 	handler *webapi.OutgoingConnectorHandler,
 	idGenerator func() string,
 	opts ...func(*Compute),
 ) *Compute {
+	if config.NumWorkers == 0 {
+		config.NumWorkers = defaultNumWorkers
+	}
 	var (
 		lggr    = logger.Named(log, "CustomCompute")
 		labeler = custmsg.NewLabeler()
 		compute = &Compute{
+			stopCh:                   make(services.StopChan),
 			log:                      lggr,
 			emitter:                  labeler,
 			registry:                 registry,
 			modules:                  newModuleCache(clockwork.NewRealClock(), 1*time.Minute, 10*time.Minute, 3),
 			transformer:              NewTransformer(lggr, labeler),
 			outgoingConnectorHandler: handler,
 			idGenerator:              idGenerator,
+			queue:                    make(chan request),
```

Comment:
I left this as non-blocking so that we backpressure onto the engine itself; if we don't succeed after 2 minutes we'll interrupt the request altogether.